src/planwise/client/components/common2.cljs

```diff
@@ -29,3 +31,17 @@
   (accountant/navigate! route)
   [ui/fixed-width (nav-params)
    [:p "Loading..."]])
+
+(defn text-field
+  [{:keys [label value] :as props}]
+  (let [focus  (r/atom false)
+        id     (str (random-uuid))]
+    (fn [{:keys [label value] :as props}]
+      [:div.mdc-text-field.mdc-text-field--upgraded {:class (when @focus "mdc-text-field--focused")}
```

Comment:
Is the `--upgraded` class necessary? It was my understanding that it only means the component has been modified/hooked by javascript code.